Addon-a11y: Reorder color blindness types by most common

maya opened the pull request:
Issue: #14767 

## What I did

- Reordered the list of color blindness types from most common to least common and grouped by type. [See some statistics here](https://www.color-blindness.com/types-of-color-blindness/).
- Changed label of mono to grayscale to more accurately reflect the filter and reduce the ambiguity that the current label suggests
- Change the name of the component and language to be more inclusive

See further discussion in the [web a11y Slack](https://web-a11y.slack.com/archives/C042TSFGN/p1619749258466500?thread_ts=1619746108.462600&cid=C042TSFGN)


<!--

Everybody: Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `master` branch as part of the release process, so you shouldn't need to worry about this.

Maintainers: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

-->

maya committed:
Reorder color blindness types by most common

shilman reviewed (approved):
Nice one @maya! Thank you 🙏x💯

maya committed:
Update order so -opia and -anomalies are together 

https://web-a11y.slack.com/archives/C042TSFGN/p1619749258466500?thread_ts=1619746108.462600&cid=C042TSFGN

maya commented:
Would you be open to changing the name of `mono` ➡️  `grayscale`? Mono makes it sound like it's a real deficiency but it's not and grayscale may be more accurate.

maya committed:
Change label name from mono to grayscale

maya committed:
Merge remote-tracking branch 'upstream/next' into next

maya committed:
Change name of a11y component to be more inclusive

shilman commented:
@maya AFAIK mono => grayscale is not a breaking change, so I think it's a great change.
